### PR TITLE
go 1.15

### PR DESCRIPTION
**Pull request by chenrui333**
As a CLI app, I think it is all good to use the latest go release. Thus using this PR to bump the go version.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -2,28 +2,27 @@ name: Go
...
     steps:
 
-    - name: Set up Go 1.14
+    - uses: actions/checkout@v2
+
+    - name: Set up Go 1.15
       uses: actions/setup-go@v1
       with:
-        go-version: 1.14
-      id: go
```
Comment (chenrui333) [inline .github/workflows/go.yml:20]: no need to use id, as it is not referenced in the other steps.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -2,28 +2,27 @@ name: Go
...
-    branches: [ master ]
+    branches: master
   pull_request:
-    branches: [ master ]
+    branches: master
 
 jobs:
-
   build:
-    name: Build
```
Comment (chenrui333) [inline .github/workflows/go.yml:12]: remove this would use job-id as job name.

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -2,28 +2,27 @@ name: Go
...
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
-
-    - name: Get dependencies
-      run: curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.24.0
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v2
+      with:
+        # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
+        version: v1.29
```
Comment (chenrui333) [inline .github/workflows/go.yml:25]: I think golangci-lint action is better than the script.

**Comment by mbrt** [comment]
Looks like a nice improvement, thank you!

**Pull request merged**